src/base/bittorrent/torrentimpl.cpp

```diff
@@ -669,6 +669,11 @@ int TorrentImpl::piecesHave() const
     return m_nativeStatus.num_pieces;
 }
 
+bool TorrentImpl::hasPieceFinished(int index) const
+{
+    return m_nativeHandle.have_piece((lt::piece_index_t)index);
```

Comment:
We disallow C-style cast.

Reply:
A static_cast should do the trick ?

Reply:
you get the best type-safety by taking the piece index as a `piece_index_t` as well. but I recognize that may be a significant commitment.

Reply:
> A static_cast should do the trick ?

Yes.

Reply:
@arvidn libtorrent types should stay in base/bittorrent ? I mean using libtorrent types in classes other than base/torrent (like gui) could break abstraction ?

Reply:
Yes. That's how qBittorrent is designed.